From 5471702f9da9ee18cd52e41fcf40b32a81d53148 Mon Sep 17 00:00:00 2001
From: Dan Liang <dan.liang@intel.com>
Date: Tue, 30 Apr 2013 14:17:10 +0800
Subject: [PORT FROM MAIN] libstagefright: Ignore stsd box flag check for MPEG4Extractor

BZ: 50109

Ignore stsd box flag check to have more MP4 files played.

Change-Id: Icbe9af5c8b1b9f046b0b8ddda9b48341261f8fae
Orig-Change-Id: I9754635f915a5fdf24b7cb8013769661876e7d73
Signed-off-by: Tianmi Chen <tianmi.chen@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
---
 media/libstagefright/MPEG4Extractor.cpp |    5 -----
 1 files changed, 0 insertions(+), 5 deletions(-)

diff --git a/media/libstagefright/MPEG4Extractor.cpp b/media/libstagefright/MPEG4Extractor.cpp
index e9a4581..103580b 100644
--- a/media/libstagefright/MPEG4Extractor.cpp
+++ b/media/libstagefright/MPEG4Extractor.cpp
@@ -1204,11 +1204,6 @@ status_t MPEG4Extractor::parseChunk(off64_t *offset, int depth) {
                 return ERROR_IO;
             }
 
-            if (U32_AT(buffer) != 0) {
-                // Should be version 0, flags 0.
-                return ERROR_MALFORMED;
-            }
-
             uint32_t entry_count = U32_AT(&buffer[4]);
 
             if (entry_count > 1) {
-- 
1.7.4.1

